src/grid/grid.directive.ts

```diff
@@ -0,0 +1,86 @@
+import { Directive, HostBinding, Input, OnInit } from '@angular/core';
+
+@Directive({
+  selector: '[ibmGrid]',
+})
+export class GridDirective implements OnInit {
+  @Input()
+  class = '';
+
+  @HostBinding('class')
+  gridClasses = 'bx--grid';
+
+  constructor() {}
+
+  ngOnInit() {
+    if (this.class !== '') {
+      this.gridClasses = `${this.gridClasses} ${this.class}`;
+    }
+  }
+}
+
+@Directive({
+  selector: '[ibmRow]',
+})
+export class RowDirective implements OnInit {
+  @Input()
+  class = '';
+
+  @HostBinding('class')
+  rowClasses = 'bx--row';
+
+  constructor() {}
+
+  ngOnInit() {
+    if (this.class !== '') {
+      this.rowClasses = `${this.rowClasses} ${this.class}`;
+    }
+  }
+}
+
+@Directive({
+  selector: '[ibmCol]',
+})
+export class ColumnDirective implements OnInit {
+  @Input()
+  class = '';
+
+  @Input()
+  columnNumbers = '{}';
+
+  @Input()
+  offsets = '{}';
+
+  @HostBinding('class')
+  columnClasses: string;
+
+  constructor() {}
+
+  ngOnInit() {
+    const classNames = [];
+    try {
+      const columns = JSON.parse(this.columnNumbers);
+      const offsets = JSON.parse(this.offsets);
```

Comment:
This does work ... but I'm wondering if asking for the columns/offsets as `[columns]="{}"` `[offsets]="{}` would be better? Then you could pass down a column/offset config as an object, without having to `JSON.stringify` it first